### PR TITLE
Create Discount & Category entity

### PR DESCRIPTION
**Pull request by VolodymyrBushko**

### CODE REVIEW

**Review by jioter — DISMISSED** [review]

File: src/main/java/com/exadel/discountwebapp/discount/entity/Discount.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    @EqualsAndHashCode.Exclude
+    @Column(name = "dis_short_description", length = 50, nullable = false)
+    private String shortDescription;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @Column(name = "dis_description", length = 510, nullable = false)
+    private String description;
+
+    @ToString.Exclude
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/discount/entity/Discount.java]: I don't see any reason why we should exclude from ToString the following fields - imageUrl, description, shortDescription

File: src/main/java/com/exadel/discountwebapp/discount/entity/Discount.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    @JoinColumn(name = "cat_id", nullable = false)
+    private Category category;
+
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToMany(cascade = {CascadeType.ALL})
+    @JoinTable(
+            name = "location_discount",
+            joinColumns = @JoinColumn(name = "ld_dis_id"),
+            inverseJoinColumns = @JoinColumn(name = "ld_loc_id")
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/discount/entity/Discount.java]: foreign keys should have the same field names as they have in their tables:
dis_id
loc_id

Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/discount/entity/Discount.java]: done

File: src/main/java/com/exadel/discountwebapp/discount/entity/Discount.java (hunk 1 above)
Comment (VolodymyrBushko) [reply src/main/java/com/exadel/discountwebapp/discount/entity/Discount.java]: done

**Review by ahlukhouski2 — APPROVED** [review]

**Review by Bozhokmaria — APPROVED** [review]

**Pull request merged**